pkg/customresourcestate/registry_factory.go

```diff
@@ -430,6 +430,10 @@ func (c compiledGauge) value(it interface{}) (*eachValue, error) {
 				Value:  0,
 			}, nil
 		}
+		// no it means no iterables were passed down, meaning that the path resolution never happened
```

Comment:
should this be moved to the beginning of the function so we can return the error early?

Reply:
That'll fail [cases](https://github.com/kubernetes/kube-state-metrics/blob/c493044e4adc1b3f45b046c61c8b334b914a0db8/pkg/customresourcestate/registry_factory_test.go#L228-L235) that expect a zero in case of `nil`.